File: bindings/python/examples/polar_line.py
from __future__ import annotations

from _shared import ExampleMeta, base_plot, polar_series, save_example

META = ExampleMeta(
    slug="polar-line",
    title="Polar line",
    summary="A polar line rendered from radius and angle vectors.",
    section="Specialized plots",
)


def build_plot():
    radius, theta = polar_series()
    return base_plot("Polar Line").polar_line(radius, theta, color="#c026d3", width=2.0)


if __name__ == "__main__":
    save_example(META, build_plot())

File: packages/ruviz/examples/polar-line.ts
import { createPlot } from "../src/index.js";
import { polarSeries } from "./data.js";
import type { PlotExampleDefinition } from "./types.js";

export const example: PlotExampleDefinition = {
  slug: "polar-line",
  title: "Polar line",
  summary: "A polar line rendered from radius and angle vectors.",
  category: "specialized",
  createPlot() {
    const { r, theta } = polarSeries();
    return createPlot().sizePx(640, 320).title("Polar Line").polarLine({ r, theta });
  },
};
